pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,463 @@
+	}
+	msg := icmp.Message{
+		Type: requestType,
+		Code: 0,
+		Body: body,
+	}
+	klog.V(4).InfoS("Send ICMP message", "IP", ip, "SeqID", seqID, "body", body)
+
+	// Serialize the ICMP message
+	msgBytes, err := msg.Marshal(nil)
+	if err != nil {
+		return err
+	}
+
+	// Send the ICMP message
+	_, err = socket.WriteTo(msgBytes, ip)
+	if err != nil {
+		return err
+	}
+
+	// Create or update the latency store
+	mutator := func(entry *NodeIPLatencyEntry) {
+		entry.LastSendTime = timeStart
+	}
+	m.latencyStore.SetNodeIPLatencyEntry(addr.String(), mutator)
+
+	return nil
+}
+
+// recvPing receives an ICMP message from the target IP address.
+func (m *NodeLatencyMonitor) recvPing(socket net.PacketConn, isIPv4 bool, stopCh <-chan struct{}) {
+	for {
+		select {
+		case <-stopCh:
+			return
+		default:
+			// max size of the ICMP message is 1500 bytes, which is the maximum size of an Ethernet frame.
+			readBuffer := make([]byte, 1500)
+			n, peer, err := socket.ReadFrom(readBuffer)
+			if err != nil {
+				if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+					klog.ErrorS(err, "Timeout reading ICMP message")
+					continue
+				}
+				klog.ErrorS(err, "Failed to read ICMP message")
+			}
+
+			destIP := peer.String()
+
+			// Parse the ICMP message
+			var msg *icmp.Message
+			if isIPv4 {
+				msg, err = icmp.ParseMessage(ProtocolICMP, readBuffer[:n])
+				if err != nil {
+					klog.ErrorS(err, "Failed to parse ICMP message")
+					continue
+				}
+				if msg.Type != ipv4.ICMPTypeEchoReply {
+					klog.V(4).InfoS("Failed to match ICMPTypeEchoReply")
+					continue
+				}
+			} else {
+				msg, err = icmp.ParseMessage(ProtocolICMPv6, readBuffer)
+				if err != nil {
+					klog.ErrorS(err, "Failed to parse ICMP message")
+					continue
+				}
+				if msg.Type != ipv6.ICMPTypeEchoReply {
+					klog.V(4).InfoS("Failed to match ICMPTypeEchoReply")
+					continue
+				}
+			}
+
+			echo, ok := msg.Body.(*icmp.Echo)
+			if !ok {
+				klog.V(4).Info("Failed to assert type as *icmp.Echo")
+				continue
+			}
+
+			klog.V(4).InfoS("Recv ICMP message", "IP", destIP, "echo", echo)
+
+			// Parse the time from the ICMP data
+			sentTime, err := time.Parse(time.RFC3339Nano, string(echo.Data))
+			if err != nil {
+				klog.ErrorS(err, "Failed to parse time from ICMP data")
+				continue
+			}
+
+			// Calculate the round-trip time
+			end := time.Now()
+			rtt := end.Sub(sentTime)
+
+			// Update the latency store
+			mutator := func(entry *NodeIPLatencyEntry) {
+				entry.LastSendTime = sentTime
+				entry.LastRecvTime = end
+				entry.LastMeasuredRTT = rtt
+			}
+			m.latencyStore.SetNodeIPLatencyEntry(destIP, mutator)
+		}
+	}
+}
+
+// pingAll sends ICMP messages to all the Nodes.
+func (m *NodeLatencyMonitor) pingAll(ipv4Socket, ipv6Socket net.PacketConn) {
+	nodeIPs := m.latencyStore.ListNodeIPs()
+	for name, toIPs := range nodeIPs {
+		for _, toIP := range toIPs {
+			klog.V(4).InfoS("Start to ping Node", "Node name", name, "Node IP", toIP)
```

Comment:
```suggestion
			klog.V(4).InfoS("Pinging Node", "name", name, "ip", toIP)
```

As a general rule, the keys for structured logs are single words. This applies to other occurrences in this file.

Reply:
OK, I have fixed it.